clustergroup/templates/applications.yaml

```diff
@@ -1,5 +1,81 @@
 {{- $namespace := cat $.Values.global.pattern $.Values.clusterGroup.name | replace " " "-" }}
 {{- range .Values.clusterGroup.applications }}
+{{- if or (.generators) (.generatorFile) (.useGeneratorValues) (.destinationURL) (.destinationNamespace) }}
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: {{ .name }}
+  namespace: {{ $namespace }}
+  labels:
+    app: {{ .name }}
+spec:
+  {{- if .generators }}
+  generators: {{ .generators | toPrettyJson }}
+  {{- else }}
+  generators:
+    - git:
+        repoURL: {{ $.Values.global.repoURL }}
+        revision: {{ $.Values.global.targetRevision }}
+        {{- if .generatorFile }}
+        files:
+          - path: {{ .generatorFile | quote }}
+        {{- end }}
+  {{- end }}
+  template:
+    metadata:
+      name: {{ coalesce .namespace $namespace }}
+    spec:
+      project: {{ .project }}
+      {{- if .syncPolicy }}
+      syncPolicy: {{ .syncPolicy | toPrettyJson }}
+      {{- else }}
+      syncPolicy:
+        automated: {}
+      {{- end }}
+      {{- if .ignoreDifferences }}
+      ignoreDifferences: {{ .ignoreDifferences | toPrettyJson }}
+      {{- end }}
+      source:
+        repoURL: {{ coalesce .repoURL $.Values.global.repoURL }}
+        targetRevision: {{ coalesce .targetRevision $.Values.global.targetRevision }} 
+        {{- if .chart }}
+        chart: {{ .chart }}
+        {{- end }}
+        {{- if .path }}
+        path: {{ .path }}
+        {{- end }}
+        {{- if .plugin }}
+        plugin: {{ .plugin }}
+        {{- end }}
+        {{- if not .kustomize }}
+        helm:
+          ignoreMissingValueFiles: true
+          valueFiles:
+            - "values.yaml"
+        {{- range .extraValueFiles }}
+            - {{ . | quote }}
+        {{- end }}
+        {{- if .useGeneratorValues }}
+          values: |-
+            {{ `{{ values }}` }}
+        {{- end }}
+          parameters:
```

Comment:
What about the other values such as `global.pattern` and `global.targetRevision` ?

Reply:
That's fair; I hadn't considered that. The values: setting here has to be a string; I'm not sure what would happen if we tried to do something super-fancy with, say, multiple YAML docs. One approach would be to add those to the parameters section (which would be super-straightforward).

Do we have  a scenario where we definitely need those parameters?